menas/src/main/scala/za/co/absa/enceladus/menas/controllers/SchemaController.scala

```diff
@@ -59,36 +59,54 @@ class SchemaController @Autowired()(
                        @RequestParam format: Optional[String]): CompletableFuture[Option[Schema]] = {
 
     val schemaType: SchemaType.Value = SchemaType.fromOptSchemaName(format)
-    val (url, fileContent, mimeType) = {
-      try {
-        val url = new URL(remoteUrl)
-        val connection = url.openConnection()
-        val mimeType = SchemaController.avscContentType // only AVSC is expected to come from the schema registry
-        val fileStream = Source.fromInputStream(connection.getInputStream)
-        val fileContent = fileStream.mkString
-        fileStream.close()
-
-        (url, fileContent, mimeType)
-
-      } catch {
-        case NonFatal(e) =>
-          throw RemoteSchemaRetrievalException(schemaType, s"Could not retrieve a schema file from $remoteUrl. Please check the correctness of the URL and a presence of the schema at the mentioned endpoint", e)
-      }
-    }
-
-    val sparkStruct = SchemaParser.getFactory(sparkMenasConvertor).getParser(schemaType).parse(fileContent)
+    val schemaResponse = schemaRegistryService.loadSchemaByUrl(remoteUrl)
+    val sparkStruct = SchemaParser.getFactory(sparkMenasConvertor).getParser(schemaType).parse(schemaResponse.fileContent)
 
     val menasFile = MenasAttachment(refCollection = RefCollection.SCHEMA.name().toLowerCase,
       refName = name,
       refVersion = version + 1,  // version is the current one, refVersion is the to-be-created one
       attachmentType = MenasAttachment.ORIGINAL_SCHEMA_ATTACHMENT,
-      filename = url.getFile,
-      fileContent = fileContent.getBytes,
-      fileMIMEType = mimeType)
+      filename = schemaResponse.url.getFile,
+      fileContent = schemaResponse.fileContent.getBytes,
+      fileMIMEType = schemaResponse.mimeType)
 
     uploadSchemaToMenas(principal.getUsername, menasFile, sparkStruct, schemaType)
   }
 
+  @PostMapping(Array("/topic"))
+  @ResponseStatus(HttpStatus.CREATED)
+  def handleTopicStem(@AuthenticationPrincipal principal: UserDetails,
+                      @RequestParam topicStem: String,
+                      @RequestParam mergeWithKey: Boolean,
+                      @RequestParam version: Int,
+                      @RequestParam name: String,
+                      @RequestParam format: Optional[String]): CompletableFuture[Option[Schema]] = {
+
+    val schemaType: SchemaType.Value = SchemaType.fromOptSchemaName(format)
+
+    val valueSchemaResponse = schemaRegistryService.loadSchemaByTopicName(s"$topicStem-value")
+    val valueSparkStruct = SchemaParser.getFactory(sparkMenasConvertor).getParser(schemaType).parse(valueSchemaResponse.fileContent)
+
+    val combinedStructType = if (mergeWithKey) {
+      val keySchemaResponse = schemaRegistryService.loadSchemaByTopicName(s"$topicStem-key")
+      val keySparkStruct = SchemaParser.getFactory(sparkMenasConvertor).getParser(schemaType).parse(keySchemaResponse.fileContent)
+
+      SchemaUtils.combineStructTypes(valueSparkStruct, keySparkStruct)
```

Comment:
Maybe the key goes first?

Reply:
Well, that depends. The task definition states 
> _An optional key subject might be specified (e.g. topic_name-key). If it is specified, the schema of the key should be merged with the value schema._

I understand it in the sense that if there is a name-clash (e.g. `field123` is contained in both key and value schema), the value's column should take precedence in case we cannot merge them somehow (both being struct). This behavior is reflected in the parameter order when calling `SchemaUtils.combineStructTypes`. It is quite easy to switch it for the key fields to take precedence. Should we switch so that key fields take precedence instead?

Reply:
Or we could make it changeable via a configuration property (which topic should take precedence on name clashes (`-key` vs `-value`)) - that would be the most elegant solution.

Provided that both scenarios actually make sense, which I am not completely sure of THB.

Reply:
After a discussion with @kevinwallimann and @yruslan, the combine-functionality was removed.